x-pack/plugins/cases/server/authorization/audit_logger.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OperationDetails } from '.';
+import { AuditLogger, EventCategory, EventOutcome } from '../../../security/server';
+
+enum AuthorizationResult {
+  Unauthorized = 'Unauthorized',
+  Authorized = 'Authorized',
+}
+
+export class AuthorizationAuditLogger {
+  private readonly auditLogger?: AuditLogger;
+
+  constructor(logger: AuditLogger | undefined) {
+    this.auditLogger = logger;
+  }
+
+  private createMessage({
+    result,
+    owner,
+    operation,
+  }: {
+    result: AuthorizationResult;
+    owner?: string;
+    operation: OperationDetails;
+  }): string {
+    const ownerMsg = owner == null ? 'of any owner' : `with "${owner}" as the owner`;
+    /**
+     * This will take the form:
+     * `Unauthorized to create case with "securitySolution" as the owner`
+     * `Unauthorized to find cases of any owner`.
+     */
+    return `${result} to ${operation.verbs.present} ${operation.docType} ${ownerMsg}`;
+  }
+
+  private logSuccessEvent({
+    message,
+    operation,
+    username,
+  }: {
+    message: string;
+    operation: OperationDetails;
+    username?: string;
+  }) {
+    this.auditLogger?.log({
+      message: `${username ?? 'unknown user'} ${message}`,
+      event: {
+        action: operation.action,
+        category: EventCategory.DATABASE,
```

Comment:
Do you think we should create `EventCategory.AUTHORIZATION` and use that?

Reply:
There's no official `authorization` field in the ecs spec currently: https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html I believe we'd want whatever we use to be present there. We could bring it up to the ECS team though.

@legrego should we use `DATABASE` for our category in the Authorization audit logs?

Reply:
Thanks Larry!

Reply:
I see! I wasn't aware of the ECS mapping. Thank you both!